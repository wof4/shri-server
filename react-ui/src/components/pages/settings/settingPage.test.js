import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { toBeInTheDocument } from '@testing-library/jest-dom';
import store from '../../../redux/reduxStore';

import SettingPage from './settingPage';

describe('Компонент: SettingPage', () => {
    it('компонент settingPage рендерится корректно', () => {
        const { container } = render(<Provider store={store}>
            <SettingPage />
        </Provider>);
        expect(container).toMatchSnapshot();
    });

    it('компонент', () => {
        const { getByText, getByTestId } = render(<Provider store={store}>
            <SettingPage />
        </Provider>);
        expect(getByText('School CI server')).toBeInTheDocument();
        expect(getByText('Settings')).toBeInTheDocument();
        expect(getByText('GitHub repository')).toBeInTheDocument();
        expect(getByText('Build command')).toBeInTheDocument();
        expect(getByText('Main branch')).toBeInTheDocument();
        expect(getByTestId('repoName')).toBeInTheDocument();
        expect(getByTestId('buildCommand')).toBeInTheDocument();
        expect(getByTestId('mainBranch')).toBeInTheDocument();
        expect(getByTestId('button-add-settings')).toBeInTheDocument();
        expect(getByTestId('button-cansel')).toBeInTheDocument();
    });
});
