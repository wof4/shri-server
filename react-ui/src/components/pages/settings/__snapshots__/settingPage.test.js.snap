// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент: SettingPage компонент settingPage рендерится корректно 1`] = `
<div>
  <div
    class="page_wrapper"
  >
    <div>
      <div
        class="header"
      >
        <div
          class="header__description"
        >
          School CI server
        </div>
      </div>
      <div>
        <div>
          <form
            action="#"
          >
            <div
              class="form_container"
            >
              <div
                class="form__title"
              >
                Settings
              </div>
              <div
                class="form__desc"
              >
                Configure repository connection and synchronization settings.
              </div>
              <div
                class="form__inputs"
              >
                <div
                  class="input_container"
                >
                  <div
                    class="form__input_title"
                  >
                    <div>
                      GitHub repository
                    </div>
                    <span
                      class="form__star_icon"
                    >
                      <img
                        alt="star"
                        src="star.png"
                      />
                    </span>
                  </div>
                  <label
                    class="form__price_input"
                    for="repoName"
                  >
                    <input
                      class="form_input"
                      data-testid="repoName"
                      name="repoName"
                      placeholder="user-name/repo-name"
                      type="text"
                      value=""
                    />
                  </label>
                </div>
                <div
                  class="input_container"
                >
                  <div
                    class="form__input_title"
                  >
                    <div>
                      Build command
                    </div>
                    <span
                      class="form__star_icon"
                    >
                      <img
                        alt="star"
                        src="star.png"
                      />
                    </span>
                  </div>
                  <label
                    class="form__price_input"
                    for="buildCommand"
                  >
                    <input
                      class="form_input"
                      data-testid="buildCommand"
                      name="buildCommand"
                      placeholder="npm ci && npm run build"
                      type="text"
                      value=""
                    />
                  </label>
                </div>
                <div
                  class="input_container"
                >
                  <img
                    alt="icon"
                    class="form__icon"
                    role="presentation"
                    src="cancel.png"
                  />
                  <div
                    class="form__input_title"
                  >
                    Main branch
                  </div>
                  <label
                    class="form__price_input"
                    for="mainBranch"
                  >
                    <input
                      class="form_input"
                      data-testid="mainBranch"
                      name="mainBranch"
                      placeholder="master |"
                      type="text"
                      value="master |"
                    />
                  </label>
                </div>
                <div
                  class="input_container_time"
                >
                  <div
                    class="form__input_title"
                  >
                    <div>
                      Synchronize every
                    </div>
                  </div>
                  <label
                    class="form__time_label"
                    for="period"
                  >
                    <input
                      class="form_input"
                      data-testid="period"
                      name="period"
                      placeholder="period"
                      type="text"
                      value="0"
                    />
                  </label>
                  <div
                    class="form__input_title"
                  >
                    minutes
                  </div>
                </div>
              </div>
              <div
                class="form__buttons"
              >
                <div
                  class="form__button"
                  data-testid="button-add-settings"
                  role="presentation"
                >
                  Save
                </div>
                <div
                  class="form__button_cansel"
                  data-testid="button-cansel"
                  role="presentation"
                >
                  Cansel
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
